[#49076] Add technical documentation for file storage integration

Kharonus opened the pull request:
- https://community.openproject.org/work_packages/49076
- added check list for new file storage integration candidates

Kharonus committed:
[#49076] Add technical documentation for file storage integration

- https://community.openproject.org/work_packages/49076
- added check list for new file storage integration candidates

Kharonus commented:
@wielinde please definitely check, if the location of the checklist is okay, or if we should merge it into any existing docs.

wielinde commented on docs/development/file-storage-integration/README.md:
```diff
@@ -0,0 +1,23 @@
+# File storage integration
```
```suggestion
# Integrating more external file storage providers
```

Kharonus replied:
added

wielinde commented on docs/development/file-storage-integration/README.md:
```diff
@@ -0,0 +1,23 @@
+# File storage integration
+
+External file storages can be used from within OpenProject with the file storage module. This adds certain use cases,
```
```suggestion
External file storages can be integrated with OpenProject via the _File storages_ module. This adds certain use cases,
```

Kharonus replied:
added

wielinde commented on docs/development/file-storage-integration/README.md:
```diff
@@ -0,0 +1,23 @@
+# File storage integration
+
+External file storages can be used from within OpenProject with the file storage module. This adds certain use cases,
+e.g. in the files tab of a work package, to interact with files located on the storage and link them to entities in
```
```suggestion
e.g. in the _Files_ tab of a work package, to interact with files located on the storage and link them to work packages in
```

Kharonus replied:
added

wielinde reviewed (changes requested):
@Kharonus Looks good to me so far. Maybe we could add some differentiation to attachments and their direct uploads, explaining that this is not about that? Also please check my inline comments/suggestions.

I don't really understand what will be the title of the menu item in the docs for this page.

Kharonus commented:
The title for the menu is something, I don't know yet. This is also a reason, why I asked, if we should integrate it somewhere else. Will first add your changes.

Kharonus committed:
[#49076] amended documentation with comments from PR

Kharonus commented:
@wielinde Can we finish that list up? Open question is currently, where to put that list.

wielinde committed:
Rewrite/extend file storages development README

- Linking the landing page of the OpenProject-:Nextcloud integration instead of Nextcloud.
- Tell community developers to reach out to the OpenProject development team

wielinde commented:
@Kharonus I added a couple of changes. If you are fine with them, please go ahead and merge.
